pkg/core/runtime/component/finalizer.go

```diff
@@ -0,0 +1,40 @@
+package component
+
+import (
+	"sync"
+
+	"github.com/kumahq/kuma/pkg/util/channels"
+)
+
+// Finalizer is a helper for implementing GracefulComponent
+type Finalizer struct {
```

Comment:
Can we use `WaitGroup` instead of `Finalizer`? Instead `Running()` you'll call `Add(1)`, instead `WaitForDone` you'll call `Wait()`

Reply:
duh! Thank you!